### PR TITLE
Facebook - get timestamp from tahoe if missing.

### PR DESCRIPTION
**Pull request by aviperes**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x ] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x ] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [ ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x ] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information
For post live videos, timestamp was not extracted.

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

### CODE REVIEW

File: youtube_dl/extractor/facebook.py [hunk 1]
```diff
@@ -424,15 +442,33 @@ def extract_from_jsmods_instances(js_data):
         uploader = clean_html(get_element_by_id(
             'fbPhotoPageAuthorName', webpage)) or self._search_regex(
             r'ownerName\s*:\s*"([^"]+)"', webpage, 'uploader',
-            fatal=False) or self._og_search_title(webpage, fatal=False)
+            fatal=False, default=None) or self._og_search_title(webpage, fatal=False, default=None) or self._search_regex(
+            r'\"ownerName\":"(.*?)"', tahoe_secondary_data,
```
Comment (dstftw) [inline youtube_dl/extractor/facebook.py]: Do not capture empty strings.

File: youtube_dl/extractor/facebook.py [hunk 2]
```diff
@@ -424,15 +442,33 @@ def extract_from_jsmods_instances(js_data):
...
             r'<abbr[^>]+data-utime=["\'](\d+)', webpage,
+            'timestamp', default=None) or self._search_regex(
+            r'data-utime=\\\"(\d+)\\\"', tahoe_secondary_data,
             'timestamp', default=None))
+
+        uploader_id = self._search_regex(
+            r'ownerid:"([\d]+)', webpage,
+            'uploader_id', default=None) or self._search_regex(
+            r'\"ownerid\":"(\d+)"', tahoe_secondary_data,
+            'uploader_id', default=None)
```
Comment (dstftw) [inline youtube_dl/extractor/facebook.py]: Relax regexes.

File: youtube_dl/extractor/facebook.py [hunk 3]
```diff
@@ -222,6 +222,10 @@ class FacebookIE(InfoExtractor):
         'params': {
             'skip_download': True,
         },
+    }, {
+        # no timestamp
+        'url': 'https://www.facebook.com/ChickenShow1996/videos/2289288568020072/',
+        'only_matching': True,
```
Comment (dstftw) [inline youtube_dl/extractor/facebook.py]: Add a complete test for what this PR is supposed to fix.

File: youtube_dl/extractor/facebook.py (hunk 2 above)
Comment (dstftw) [reply youtube_dl/extractor/facebook.py]: Allow arbitrary whitespace and both quote types.

File: youtube_dl/extractor/facebook.py [hunk 4]
```diff
@@ -423,16 +452,35 @@ def extract_from_jsmods_instances(js_data):
             video_title = 'Facebook video #%s' % video_id
         uploader = clean_html(get_element_by_id(
             'fbPhotoPageAuthorName', webpage)) or self._search_regex(
-            r'ownerName\s*:\s*"([^"]+)"', webpage, 'uploader',
-            fatal=False) or self._og_search_title(webpage, fatal=False)
+            r'ownerName\s*:\s*"([^"]+)"', webpage, 'uploader',default=None) or \
+                   self._og_search_title(webpage, default=None) or self._search_regex(
+                        r'\"ownerName\":"(.+?)"', tahoe_secondary_data,
```
Comment (dstftw) [inline youtube_dl/extractor/facebook.py]: This breaks if `tahoe_secondary_data` request has failed.

Comment (aviperes) [reply youtube_dl/extractor/facebook.py]: Thanks. I added empty string as default.

**Comment by aviperes** [comment]
All comments was fixed. Any more comments?

**Comment by aviperes** [comment]
All comments was fixed. Is there any other comments ?

**Comment by aviperes** [comment]
Hi, I improved PR. What is stopping us from merging this PR?

**Pull request closed without merging**